lib/tasks/vmodl_helper.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+require 'active_support/core_ext/enumerable'
+require 'active_support/inflector'
+require 'optimist'
+require 'pathname'
+require 'wsdl/parser'
+require 'rbvmomi'
+require 'rbvmomi/pbm'
+require 'rbvmomi/sms'
+
+class VmodlHelper
+  class << self
+    def verify!(argv)
+      opts = Optimist.options(argv) do
+        educate_on_error
+        opt :wsdl, 'Path to the vsphere-ws wsdl file', type: :string, required: true
+        opt :vmodl, 'Path to the vmodl.db', type: :string, default: 'vmodl.db'
+        banner <<~EOS
+          Usage:
+          rake vmodl:verify -- --wsdl=path/to/wsdl
+        EOS
+      end
+
+      new(vmodl_path: opts[:vmodl], wsdl_path: opts[:wsdl]).verify!
+    end
+
+    def generate!(argv)
+      opts = Optimist.options(argv) do
+        educate_on_error
+        opt :wsdl, 'Path to the vsphere-ws wsdl file', type: :string, required: true
+        opt :vmodl, 'Path to the vmodl.db', type: :string, default: 'vmodl.db'
+        banner <<~EOS
+          Usage:
+          rake vmodl:generate -- --wsdl=path/to/wsdl
+        EOS
+      end
+
+      new(vmodl_path: opts[:vmodl], wsdl_path: opts[:wsdl]).generate!
+    end
+  end
```

Comment:
There some redundancy here that I think can be cleaned up.
```suggestion
  class << self
    private def run!(mode, argv)
      opts = Optimist.options(argv) do
        educate_on_error
        opt :wsdl, 'Path to the vsphere-ws wsdl file', type: :string, required: true
        opt :vmodl, 'Path to the vmodl.db', type: :string, default: 'vmodl.db'
        banner <<~EOS
          Usage:
          rake vmodl:#{mode} -- --wsdl=path/to/wsdl
        EOS
      end
      new(vmodl_path: opts[:vmodl], wsdl_path: opts[:wsdl]).send("#{mode}!")
    end

    def verify!(argv)
      run!(:verify, argv)
    end

    def generate!(argv)
      run!(:generate, argv)
    end
  end
```